### PR TITLE
Replace data_frame() by tibble()

### PR DESCRIPTION
**Pull request by Copepoda**
Changed data_frame() to tibble() in dplyr from issue 4096

Original issue:
This issue is to be dealt with on Tidyverse developer day 2019. 🙏 don't submit pull requests for it outside of the event. We are of course interested in having this fixed, but much more interested in doing it as part of the developer day

  identify uses of the deprecated function data_frame() in the dplyr codebase: code, docs, tests, vignette.
  replace with uses of tibble()

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -938,7 +938,7 @@ Functions related to the creation and coercion of `tbl_df`s, now live in their o
   `as_data_frame()`), data frames (trivial), and matrices (with efficient
   C++ implementation) (#876). It no longer strips subclasses.
 
-* The internals of `data_frame()` and `as_data_frame()` have been aligned,
+* The internals of `tibble()` and `as_data_frame()` have been aligned,
```
Comment (romainfrancois) [inline NEWS.md]: We should not make these changes in that file, especially for sections that refer to old releases, can you revert changes in this file please ?

Comment (Copepoda) [reply NEWS.md]: Can do. :)

Comment (romainfrancois) [reply NEWS.md]: 👍

**Comment by romainfrancois** [comment]
Great. I just noticed that we are in a similar situation with `as_data_frame()` being deprecated and replaced by `as_tibble()`. 

I can open another issue, but it feels like we could tackle it here since it's very similar

**Comment by Copepoda** [comment]
> Great. I just noticed that we are in a similar situation with `as_data_frame()` being deprecated and replaced by `as_tibble()`.
> 
> I can open another issue, but it feels like we could tackle it here since it's very similar

done 👍

**Comment by romainfrancois** [comment]
Thanks. That looks good. I'll merge this once we get ✅ from travis.

File: tests/testthat/test-equality.r [hunk 2]
```diff
@@ -78,7 +78,7 @@ test_that("factor comparison requires strict equality of levels (#2440)", {
 })
 
 test_that("BoolResult does not overwrite singleton R_TrueValue", {
-  dplyr:::equal_data_frame(mtcars, mtcars)
+  dplyr:::equal_tibble(mtcars, mtcars)
```
Comment (romainfrancois) [inline tests/testthat/test-equality.r]: There is no `equal_tibble()` function, I'm getting this error locally: 

```r
> test_check("dplyr")
── 1. Error: BoolResult does not overwrite singleton R_TrueValue (@test-equality.r#81)  ────────────────────────────────────
object 'equal_tibble' not found
1: dplyr:::equal_tibble at testthat/test-equality.r:81
2: get(name, envir = asNamespace(pkg), inherits = FALSE)

══ testthat results  ═══════════════════════════════════════════════════════════════════════════════════════════════════════
OK: 3276 SKIPPED: 9 FAILED: 1
1. Error: BoolResult does not overwrite singleton R_TrueValue (@test-equality.r#81) 

Error: testthat unit tests failed
In addition: Warning message:
call dbDisconnect() when finished working with a connection 
Execution halted
```

**Comment by romainfrancois** [comment]
closes #4096

**Comment by romainfrancois** [comment]
Thanks again @Copepoda for dealing with this issue on developer day 👍

**Comment by Copepoda** [comment]
Awesome! Dev day was so much fun. :)

**Pull request merged**